packages/expo-asset/src/EmbeddedAssets.ts

```diff
@@ -1,17 +1,27 @@
 import Constants from 'expo-constants';
 import * as FileSystem from 'expo-file-system';
+import * as Updates from 'expo-updates';
 
 // Fast lookup check if assets are available in the local bundle
 const bundledAssets = new Set(FileSystem.bundledAssets || []);
+const localAssets = Updates.localAssets;
 
 /**
  * Returns the local URI of an embedded asset from its hash and type, or null if the asset is not
  * included in the app bundle.
  */
 export function getEmbeddedAssetUri(hash: string, type: string | null): string | null {
-  const assetName = `asset_${hash}${type ? `.${type}` : ''}`;
-  if (__DEV__ || Constants.appOwnership !== 'standalone' || !bundledAssets.has(assetName)) {
+  if (__DEV__) {
     return null;
   }
-  return `${FileSystem.bundleDirectory}${assetName}`;
+  if (!localAssets.hasOwnProperty(hash)) {
```

Comment:
IIRC the hash that the bundler includes in the JS bundle is an MD5 hash, while expo-updates uses a hashing algorithm that depends on the asset (currently just SHA-1 but could be something else on a per-asset basis)

Reply:
the key in the `localAssets` object is actually the filename from the remote URL (meaning in practice it will be the MD5 hash). I've done some refactoring/renaming to make this a little clearer.